ad9959/src/lib.rs

```diff
@@ -511,6 +491,72 @@ impl<I: Interface> Ad9959<I> {
     }
 }
 
+/// Validate the internal system clock configuration of the chip.
+///
+/// Arguments:
+/// * `reference_clock_frequency` - The reference clock frequency provided to the AD9959 core.
+/// * `multiplier` - The frequency multiplier of the system clock. Must be 1 or 4-20.
+///
+/// Returns:
+/// The system clock frequency to be configured.
+pub fn validate_clocking(
+    reference_clock_frequency: f32,
+    multiplier: u8,
+) -> Result<f32, Error> {
+    if multiplier != 1 && !(4..=20).contains(&multiplier)
+        || (multiplier != 1 && reference_clock_frequency < 10e6)
+        || reference_clock_frequency < 1e6
+    {
+        return Err(Error::Bounds);
+    }
+
+    let frequency = multiplier as f32 * reference_clock_frequency;
+    if !(255e6..=500e6).contains(&frequency)
+        && !(100e6..=160e6).contains(&frequency)
+        && (multiplier != 1 || !(0.0..100e6).contains(&frequency))
```

Comment:
Similar to above - please break this out into multiple conditionals and document why each conditional is required. Also, break the magic numbers out into named constants at the file-scope so it's clear what they represent (and potentially could be combined).